ebay-select: TypeScript compilation error with onChange

When EbaySelect with onChange used with TypeScript, I got compilation error due to conflicting types:

```
error TS2322: Type '(e: ChangeEvent<HTMLSelectElement>, selectedIndex: number, newValue: string | number | readonly string[]) => void' is not assignable to type 'ChangeEventHandler<HTMLSelectElement> & ((e: ChangeEvent<HTMLSelectElement>, selectedIndex: number, newValue: SelectValue) => void)'.
  Type '(e: ChangeEvent<HTMLSelectElement>, selectedIndex: number, newValue: string | number | readonly string[]) => void' is not assignable to type 'ChangeEventHandler<HTMLSelectElement>'.

59         onChange={handleSelectChange}
           ~~~~~~~~

  node_modules/@types/react/index.d.ts:2375:9
    2375         onChange?: ChangeEventHandler<T> | undefined;
                 ~~~~~~~~
    The expected type comes from property 'onChange' which is declared here on type 'IntrinsicAttributes & ClassAttributes<HTMLSelectElement> & SelectHTMLAttributes<HTMLSelectElement> & { ...; } & { ...; }'
```

## How to reproduce

- Use any TypeScript (tested with 3.9 and 4.6) with the latest type definitions for react v16.14.30.
- Add EbaySelect with onChange

```tsx
const handleSelectChange: (
  e: React.ChangeEvent<HTMLSelectElement>,
  selectedIndex: number,
  newValue: string | ReadonlyArray<string> | number
) => void = (event, selectedIndex, newValue) => {
  // ....
}

// ...
<EbaySelect
  name="endReason"
  onChange={handleSelectChange}
>
  {benefits.map((benefit, index) => 
    <EbaySelectOption key={index} value={benefit.value}>
      {benefit.title}
    </EbaySelectOption>
  )}
</EbaySelect>
```

## Cause

This is due to conflicting types between:

- HTML select onChange with type `ChangeEventHandler<HTMLSelectElement>` (1 parameter)
- EbaySelect onChange with type `(e: ChangeEvent<HTMLSelectElement>, selectedIndex: number, newValue: string | number | readonly string[]) => void` (3 parameters)

EbaySelect used all select props and try to overwrite its onChange: https://github.com/eBay/ebayui-core-react/blob/e13c4e6f9beba01ce19ba8c2c94d77c3dec57357/src/ebay-select/ebay-select.tsx#L11-L18

We should omit onchange from EbaySelect before overwriting it. See https://react-typescript-cheatsheet.netlify.app/docs/advanced/patterns_by_usecase/#props-omit-prop-from-a-type

## Workaround

Cast the handler with ChangeEventHandler<HTMLSelectElement>:

```tsx
<EbaySelect
  name="endReason"
  onChange={handleSelectChange as React.ChangeEventHandler<HTMLSelectElement>}
>
```

--- a/src/ebay-select/__tests__/index.stories.tsx
+++ b/src/ebay-select/__tests__/index.stories.tsx
@@ -1,4 +1,4 @@
-import React, { useState } from 'react'
+import React, { ChangeEvent, useState } from 'react';
 import { storiesOf } from '@storybook/react'
 import { action } from '../../../.storybook/action'
 import { EbaySelect, EbaySelectOption, EbayButton } from '../../index'
@@ -92,7 +92,9 @@ storiesOf(`ebay-select`, module)
     .add(`Controlled component`, () => {
         const TestControlledComponent = () => {
             const [selectedValue, setSelectedValue] = useState('UK')
-            const handleChange = (e) => setSelectedValue(e.target.value)
+            const handleChange = (e: ChangeEvent<HTMLSelectElement>, selectedIndex: number, newValue: string) => {
+                setSelectedValue(newValue)
+            }
 
             return (
                 <div style={{ width: `500px` }}>

--- a/src/ebay-select/ebay-select.tsx
+++ b/src/ebay-select/ebay-select.tsx
@@ -8,7 +8,7 @@ import { useFloatingLabel } from '../ebay-floating-label/hooks'
 const isControlled = value => typeof value !== 'undefined'
 
 type SelectValue = string | ReadonlyArray<string> | number;
-export type EbaySelectProps = ComponentProps<'select'> & {
+export type EbaySelectProps = Omit<ComponentProps<'select'>, 'onChange'> & {
     borderless?: boolean;
     defaultValue?: SelectValue;
     onChange?: (e: ChangeEvent<HTMLSelectElement>, selectedIndex: number, newValue: SelectValue) => void;

